src/ExternalSearch.Providers.VatLayer/VatLayerExternalSearchProvider.cs

```diff
@@ -374,6 +376,59 @@ public IPreviewImage GetPrimaryEntityPreviewImage(ExecutionContext context, IExt
             }
         }
 
+
+        public ConnectionVerificationResult VerifyConnection(ExecutionContext context, IReadOnlyDictionary<string, object> config)
+        {
+            IDictionary<string, object> configDict = config.ToDictionary(entry => entry.Key, entry => entry.Value);
+            var jobData = new VatLayerExternalSearchJobData(configDict);
+
+            var vat = WebUtility.UrlEncode("GB765970776");
+            var client = new RestClient("http://www.apilayer.net/api");
+            var request = new RestRequest($"validate?access_key={jobData.ApiToken}&vat_number={vat}&format=1", Method.GET);
+
+            var response = client.ExecuteAsync<VatLayerResponse>(request).Result;
+
+            return ConstructVerifyConnectionResponse(response);
+        }
+
+        private ConnectionVerificationResult ConstructVerifyConnectionResponse(IRestResponse<VatLayerResponse> response)
+        {
+            var isSuccessResponse = response.IsSuccessful;
+            var errorMessageBase = $"{Constants.ProviderName} returned \"{(int)response.StatusCode} {response.StatusDescription}\".";
+            if (response.ErrorException != null)
+            {
+                return new ConnectionVerificationResult(false, $"{errorMessageBase} {(!string.IsNullOrWhiteSpace(response.ErrorException.Message) ? response.ErrorException.Message : "This could be due to breaking changes in the external system")}.");
+            }
+
+            var responseData = response.Data;
+            if (responseData == null || (responseData != null && !responseData.Valid))
+            {
+                VatLayerErrorResponse content;
+                try
+                {
+                    content = JsonConvert.DeserializeObject<VatLayerErrorResponse>(response.Content);
+                    if (!string.IsNullOrWhiteSpace(content.Error.Type) && content.Error.Type.Equals("invalid_access_key", StringComparison.OrdinalIgnoreCase) == true)
+                    {
+                        return new ConnectionVerificationResult(false, $"{Constants.ProviderName} returned \"401 Unauthorized\". This could be due to an invalid API key.");
+                    }
+                }
+                catch (Exception) {
+                }
```

Comment:
Don't leave catch blocks empty. Either handle the exception or log it.

I assume you have a catch here incase DeserializeObject throws an exception. Wouldn't we want to send a message back to the user in this case? `Server returned OK but the content received was invalid` ?

Reply:
updated